src/include/matrix.h

```diff
@@ -767,12 +761,34 @@ inline void Matrix<ELEMENT_TYPE>::pow(const size_t &exponent) {
     pow(exponent, *this);
 }
 
-// TODO
-template <class ELEMENT_TYPE>
-template <class O>
-void Matrix<ELEMENT_TYPE>::transpose(Matrix<O> &output) const {}
 template <class ELEMENT_TYPE>
-void Matrix<ELEMENT_TYPE>::transpose() {}
+Matrix<ELEMENT_TYPE> Matrix<ELEMENT_TYPE>::transpose() const {
+    // single mode
+    Matrix<> output(shape(), 0);
```

Comment:
remove the `, 0`. In our new version, `Matrix<> output(shape())` will create a new matrix whose elements are all `ELEMENT_TYPE()`.